core_impl/src/main/java/com/google/instrumentation/stats/StatsManagerImpl.java

```diff
@@ -13,21 +13,88 @@
 
 package com.google.instrumentation.stats;
 
+import com.google.common.collect.Multimap;
+import com.google.instrumentation.common.Timestamp;
+import com.google.instrumentation.stats.View.DistributionView;
+import com.google.instrumentation.stats.View.IntervalView;
+import com.google.instrumentation.stats.ViewDescriptor.DistributionViewDescriptor;
+import com.google.instrumentation.stats.ViewDescriptor.IntervalViewDescriptor;
+import java.util.Collection;
+
 /**
  * Native Implementation of {@link StatsManager}.
  */
 public final class StatsManagerImpl extends StatsManager {
 
+  private final Multimap<MeasurementDescriptor.Name, View> measurementDescriptorToViewMap =
+      MeasurementDescriptorToViewMap.getMap();
   private final StatsContextFactoryImpl statsContextFactory = new StatsContextFactoryImpl();
 
   @Override
   public void registerView(ViewDescriptor viewDescriptor) {
-    throw new UnsupportedOperationException("Not implemented yet.");
+    // We are using a preset measurement {@link RpcConstants.RPC_CLIENT_ROUNDTRIP_LATENCY}
+    // and view {@link RpcConstants.RPC_CLIENT_ROUNDTRIP_LATENCY_VIEW} for this prototype.
+    // The prototype does not allow setting measurement descriptor entries dynamically for now.
+    // TODO(songya): remove the logic for checking preset descriptor.
+    if (!viewDescriptor.equals(RpcConstants.RPC_CLIENT_ROUNDTRIP_LATENCY_VIEW)) {
+      throw new UnsupportedOperationException(
+          "The prototype will only support Distribution View RPC_CLIENT_ROUNDTRIP_LATENCY_VIEW.");
+    }
+
+    MeasurementDescriptor measurementDescriptor = viewDescriptor.getMeasurementDescriptor();
+    Collection<View> views = measurementDescriptorToViewMap.get(
+        measurementDescriptor.getMeasurementDescriptorName());
+
+    if (views != null) {
+      for (View view : views) {
+        if (view.getViewDescriptor().equals(viewDescriptor)) {
+          throw new IllegalArgumentException("View has already been registered.");
```

Comment:
I just thought that the API would be easier to use if client code didn't have to ensure that views were only added once.